File: _Inventarios/DaxInventarios_org/medidasArticulos.cs
using System;
using System.Data;
using System.Data.SqlClient;

namespace DaxInvent
{
    public class medidasArticulos
    {
        public double MulMedida(string DeMed, string AMed, string strsys)
        {
            if ((DeMed ?? "") == (AMed ?? "") | string.IsNullOrEmpty(DeMed) | string.IsNullOrEmpty(AMed))
                return 1d;
            var Medida = new DataTable();
            var da = new SqlDataAdapter("select isnull(Cnv_Multiplo,1) as Cnv_Multiplo from conversion where Cnv_DeMedida = '" + DeMed + "' and Cnv_Amedida = '" + AMed + "'", strsys);
            using (Medida)
            using (da)
            {
                da.Fill(Medida);
                if (Medida.Rows.Count == 0)
                    return 1d;
                return Convert.ToDouble(Medida.Rows[0]["Cnv_Multiplo"]);
            }
        }

        public double CambiaEmpaque(string AMed, string strdax, string strsys, string codigo = "")
        {
            if (AMed.Length == 0 || codigo.Length == 0) return 1;
            double CambiaEmpaqueRet = 0;
            string ssql;

            CambiaEmpaqueRet = 0d;
            ssql = "SELECT Art_codigo, isnull(Art_unimed,'') as art_unimed";
            ssql = ssql + ", isnull(CodEmpaque1,'') as CodEmpaque1, isnull(ValEmpaque1,0) as ValEmpaque1, isnull(CodEmpaque2,'') as CodEmpaque2, isnull(ValEmpaque2,0) as ValEmpaque2 ";
            ssql = ssql + ", isnull(CodEmpaque3,'') as CodEmpaque3, isnull(ValEmpaque3,0) as ValEmpaque3, isnull(CodEmpaque4,'') as CodEmpaque4, isnull(ValEmpaque4,0) as ValEmpaque4";
            ssql = ssql + ", isnull(CodEmpaque5,'')as CodEmpaque5, isnull(ValEmpaque5,0) as ValEmpaque5";
            ssql = ssql + " From AdcArt ";
            ssql = ssql + " where art_codigo = '" + codigo + "' ";

            using (SqlDataAdapter da = new SqlDataAdapter(ssql, strdax))
            {
                var rs = new DataTable();
                da.Fill(rs);
                if (rs.Rows.Count > 0)
                {
                    DataRow Drow = rs.Rows[0];
                    if (Drow["art_unimed"].ToString() == AMed) return 1;

                    try
                    {
                        if (AMed == Drow["codempaque1"].ToString()) CambiaEmpaqueRet = Convert.ToDouble(Drow["valempaque1"]);
                        else
                        if (AMed == Drow["codempaque2"].ToString()) CambiaEmpaqueRet = Convert.ToDouble(Drow["valempaque2"]);
                        else
                        if (AMed == Drow["codempaque3"].ToString()) CambiaEmpaqueRet = Convert.ToDouble(Drow["valempaque3"]);
                        else
                        if (AMed == Drow["codempaque4"].ToString()) CambiaEmpaqueRet = Convert.ToDouble(Drow["valempaque4"]);
                        else
                        if (AMed == Drow["codempaque5"].ToString()) CambiaEmpaqueRet = Convert.ToDouble(Drow["valempaque5"]);
                    }catch { }
                }
            }
            if (CambiaEmpaqueRet == 0)
            {
                using (SqlDataAdapter da = new SqlDataAdapter("select isnull(caracteristica1,0) as valor from syscod where tiporeferencia = 'empaque' and Abreviación = '" + AMed + "' ", strsys))
                {
                    DataTable rs = new DataTable();
                    da.Fill(rs);
                    if (rs.Rows.Count > 0)
                    {
                        CambiaEmpaqueRet = Convert.ToDouble(rs.Rows[0]["Valor"]);
                    }
                    rs = null;
                }
            }
            return CambiaEmpaqueRet;
        }
    }
}

File: _Inventarios/VersionesIndividualesIncorporadasADaxInventarios/MantenimientoMedidas/MantenimientoMedidas/medidasArticulos.vb
Imports System.Data
Imports System.Data.SqlClient

Public Class medidasArticulos
    Public Function MulMedida(DeMed As String, AMed As String, strsys As String) As Double
        If DeMed = AMed Or DeMed = "" Or AMed = "" Then Return 1
        Dim Medida As New DataTable
        Dim da As New SqlDataAdapter("select isnull(Cnv_Multiplo,1) as Cnv_Multiplo from conversion where Cnv_DeMedida = '" + DeMed + "' and Cnv_Amedida = '" + AMed + "'", strsys)
        Using (Medida)
            Using (da)
                da.Fill(Medida)
                If Medida.Rows.Count = 0 Then Return 1
                Return CDbl(Medida.Rows(0)("Cnv_Multiplo"))
            End Using
        End Using
    End Function

    Public Function CambiaEmpaque(AMed As String, strdax As String, strsys As String, Optional codigo As String = "") As Double
        Dim rs As New DataTable
        Dim ssql As String
        On Error Resume Next
        CambiaEmpaque = 0
        ssql = "SELECT Art_codigo, isnull(Art_unimed,'') as art_unimed"
        ssql = ssql & ", isnull(CodEmpaque1,'') as CodEmpaque1, isnull(ValEmpaque1,0) as ValEmpaque1, isnull(CodEmpaque2,'') as CodEmpaque2, isnull(ValEmpaque2,0) as ValEmpaque2 "
        ssql = ssql & ", isnull(CodEmpaque3,'') as CodEmpaque3, isnull(ValEmpaque3,0) as ValEmpaque3, isnull(CodEmpaque4,'') as CodEmpaque4, isnull(ValEmpaque4,0) as ValEmpaque4"
        ssql = ssql & ", isnull(CodEmpaque5,'')as CodEmpaque5, isnull(ValEmpaque5,0) as ValEmpaque5"
        ssql = ssql & " From AdcArt "
        ssql = ssql & " where art_codigo = '" + codigo + "' "
        Dim da As New SqlDataAdapter(ssql, strdax)
        da.Fill(rs)
        If rs.Rows.Count > 0 Then
            If rs.Rows(0)("art_unimed").ToString() = AMed Then rs.Dispose() : Return 1
            With rs.Rows(0)
                Select Case AMed
                    Case .Item("codempaque1").ToString()
                        CambiaEmpaque = CDbl(.Item("valempaque1"))
                    Case .Item("codempaque2").ToString()
                        CambiaEmpaque = CDbl(.Item("valempaque2"))
                    Case .Item("codempaque3").ToString()
                        CambiaEmpaque = CDbl(.Item("valempaque3"))
                    Case .Item("codempaque4").ToString()
                        CambiaEmpaque = CDbl(.Item("valempaque4"))
                    Case .Item("codempaque5").ToString()
                        CambiaEmpaque = CDbl(.Item("valempaque5"))
                End Select
            End With
        End If
        rs.Dispose()
        If CambiaEmpaque = 0 Then
            da = New SqlDataAdapter("select isnull(caracteristica1,0) as valor from syscod where tiporeferencia = 'empaque' and Abreviación = '" & AMed & "' ", strsys)
            If rs.Rows.Count > 0 Then
                CambiaEmpaque = Val(rs.Rows(0)("Valor"))
            End If
            rs = Nothing
        End If
    End Function
End Class
